orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy

```diff
@@ -16,19 +16,119 @@
 package com.netflix.spinnaker.orca.pipeline.util
 
 import com.fasterxml.jackson.databind.ObjectMapper
-import com.netflix.spinnaker.orca.pipeline.model.Execution
 import com.netflix.spinnaker.orca.pipeline.model.Pipeline
 import com.netflix.spinnaker.orca.pipeline.model.PipelineStage
 import com.netflix.spinnaker.orca.pipeline.model.Stage
+import com.netflix.spinnaker.orca.pipeline.model.TestPipelineStage
 import org.springframework.beans.factory.annotation.Autowired
 import spock.lang.Specification
 import spock.lang.Unroll
 
+import java.util.regex.Pattern
+
 class PackageInfoSpec extends Specification {
 
   @Autowired
   ObjectMapper mapper
 
+  @Unroll
+  def "Test that the artifact matching method returns whether or not there is a match"() {
+
+    given:
+    def pattern = Pattern.compile("${input.fileName}.*")
+
+    when:
+    def result = PackageInfo.artifactMatch(input.artifacts, pattern)
+
+    then:
+    result == matchFound
+
+    where:
+    input                                                                                                     || matchFound
+    [fileName: "foo", artifacts: []]                                                                          || false
+    [fileName: "bar", artifacts: [[:]]]                                                                       || false
+    [fileName: "bat", artifacts: [[:], [:]]]                                                                  || false
+    [fileName: "testFileName", artifacts: [[fileName: "testFileName_1"]]]                                     || true
+    [fileName: "baz", artifacts: [[fileName: "badTestFileName_1"]]]                                           || false
+    [fileName: "testFileName", artifacts: [[fileName: "testFileName_1"], [fileName: "testFileName_2"]]]       || true
+    [fileName: "blah", artifacts: [[fileName: "badTestFileName_1"], [fileName: "badTestFileName_1"]]]         || false
+    [fileName: "testFileName", artifacts: [[fileName: "badTestFileName"], [fileName: "testFileName_1"]]]      || true
+    [fileName: "testFileName", artifacts: [[fileName: "testFileName_1"], [fileName: "badTestFileName"]]]      || true
+  }
+
+  def "If no artifacts in current build info or ancestor stages, code should execute as normal"() {
+
+    given:
+    def quipStage = new PipelineStage(
+      execution: Pipeline.builder().withTrigger(["buildInfo": ["artifacts": [["fileName": "testFileName"]]]]).build(),
+      context: [buildInfo: [name: "someName"], package: "testPackageName"]
+    )
+
+    def packageType = PackageType.DEB
+    def packageInfo =
+      new PackageInfo(quipStage, packageType.packageType, packageType.versionDelimiter, true, false, new ObjectMapper())
+
+    when:
+    packageInfo.findTargetPackage(false)
+
+    then:
+    def ex = thrown(IllegalStateException)
+    ex.message.startsWith("Unable to find deployable artifact starting with")
+  }
+
+  private static Map gc(String name) {
```

Comment:
Sorry ... I think there was some confusion earlier.

In this particular usage, I don't believe you need to extract this logic into a method like `gc`.

I would normally use the short alias methods when I want to tidy up the colums in `where:` block.

Since `sr()` is only called in a single location, I'd probably just inline this there.